demos/ASP.Net Core API Gateway/Divergent.Sales.ViewModelComposition/OrdersListViewModelAppender.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Divergent.Sales.ViewModelComposition.Events;
+using ITOps.ViewModelComposition;
+using ITOps.ViewModelComposition.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Divergent.Sales.ViewModelComposition
+{
+    public class OrdersListViewModelAppender : IViewModelAppender
+    {
+        public bool Matches(RouteData routeData, string verb)
+        {
+            /*
+             * matching is a bit weak in this sample, it's designed 
+             * this way to satisfy both the Gateway and the Mvc sample
+             */
+            var controller = (string)routeData.Values["controller"];
+
+            return HttpMethods.IsGet(verb)
+                && controller.ToLowerInvariant() == "orders"
+                && !routeData.Values.ContainsKey("id");
+        }
+
+        public async Task Append(dynamic vm, RouteData routeData, IQueryCollection query)
```

Comment:
That's a good point. If appenders create their own VM that is returned the issue is that who manages the appender then needs to act as a coordinator in order to merge together all the returned view models. The result is that the coordinator knows the structure and we don't want that.

The used approach on the contrary makes the coordinator totally unaware of the structure, leaving the thing with the possibility that 2 appenders clash on the same property name for example.